packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -306,8 +309,16 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// The [builder], [addAutomaticKeepAlives], [addRepaintBoundaries],
   /// [addSemanticIndexes], and [semanticIndexCallback] arguments must not be
   /// null.
+  ///
+  /// Providing a [findItemIndex] lets [SliverChildBuilderDelegate]
+  /// able to locate the children in the case of re-ordering. If re-ordering
+  /// children without providing [findItemIndex] might result in state-loss.
+  ///
+  /// [findItemIndex] will be called with an input key and should return the
+  /// child index associate with the given key, -1 if not found.
   const SliverChildBuilderDelegate(
     this.builder, {
+    this.findItemIndex,
```

Comment:
this should probably be findChildIndex (the SliverChildBuilderDelegate deals in terms of children, the PageView calls the same thing Items).